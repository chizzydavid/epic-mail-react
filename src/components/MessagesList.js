import React from 'react'
import PropTypes from 'prop-types';
import MessageItem from './MessageItem';

const MessageList = (props) => {
  const { 
    isLoading, messages, feedback, errors
  } = props;

  const formFeedback = feedback ? feedback : errors.error ? errors.error : '';


  const allMessages = messages.map(({message_id, subject, sender_id, message, photo, status}) => 
    <MessageItem 
      key={message_id}
      message_id={message_id}
      sender_id={sender_id}
      subject={subject}
      message={message}
      photo={photo}
      status={status}
    />
  )

  return (
    <main className="message-wrapper">
      <div className={isLoading ? '' : 'hide'} id="loader">
        <div className="sk-wave">
          <div className="sk-rect sk-rect1"></div>
          <div className="sk-rect sk-rect2"></div>
          <div className="sk-rect sk-rect3"></div>
          <div className="sk-rect sk-rect4"></div>
          <div className="sk-rect sk-rect5"></div>
        </div>
      </div>
      <div>
        <p className={formFeedback ? '' : isLoading ? 'hide' : ''} id="feedback">
          {formFeedback}
        </p>
      </div>

      <div className="message-container"> {allMessages} </div>
    </main>
  )
}

MessageList.propTypes = {
  messages: PropTypes.array.isRequired,
  isLoading: PropTypes.bool.isRequired,
  feedback: PropTypes.string.isRequired,
  errors: PropTypes.object.isRequired,
}

export default MessageList;
