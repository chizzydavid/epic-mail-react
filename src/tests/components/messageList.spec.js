import React from 'react';
import { shallow } from 'enzyme';
import MessagesList from '../../components/MessagesList';

function shallowSetup() {
  const props = {
    isLoading: false,
    messages: [
      {
        created_at: "July 23rd 2019, 2:47:12 am",
        message: "hey man mhey man man",
        message_id: 3,
        parent_msg_id: "0",
        photo: "default-user.png",
        sender_id: 2,
        status: "sent",
        subject: "hey chizzzy"
      },
      {
        created_at: "July 23rd 2019, 2:47:12 am",
        message: "hey man mhey man man",
        message_id: 3,
        parent_msg_id: "0",
        photo: "default-user.png",
        sender_id: 2,
        status: "sent",
        subject: "hey chizzzy"
      }
    ],
    feedback: 'no received messages',
    errors: {error: 'message not found'},
  }
  const enzymeWrapper = shallow(<MessagesList {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Shallow rendered Message List', () => {
  it('should render message list component', () => {
    const { enzymeWrapper } = shallowSetup();
    expect(enzymeWrapper).toMatchSnapshot();
  });
});

