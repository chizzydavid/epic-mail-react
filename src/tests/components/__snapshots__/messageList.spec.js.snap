// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered Message List should render message list component 1`] = `ShallowWrapper {}`;
